store/mongo/datastore_mongo.go

```diff
@@ -535,6 +535,71 @@ func (db *DataStoreMongo) UpdateDevicesGroup(
 	}, nil
 }
 
+func (db *DataStoreMongo) GetFiltersAttributes(ctx context.Context) ([]model.FilterAttribute, error) {
+	database := db.client.Database(mstore.DbFromContext(ctx, DbName))
+	collDevs := database.Collection(DbDevicesColl)
+
+	const DbCount = "count"
+
+	cur, err := collDevs.Aggregate(ctx, []bson.M{
+		{
+			"$project": bson.M{
+				"attributes": bson.M{
+					"$objectToArray": "$" + DbDevAttributes,
+				},
+			},
+		},
+		{
+			"$unwind": "$" + DbDevAttributes,
+		},
+		{
+			"$project": bson.M{
+				DbDevAttributesName:  "$" + DbDevAttributes + ".v." + DbDevAttributesName,
+				DbDevAttributesScope: "$" + DbDevAttributes + ".v." + DbDevAttributesScope,
+			},
+		},
+		{
+			"$group": bson.M{
+				DbDevId: bson.M{
+					DbDevAttributesName:  "$" + DbDevAttributesName,
+					DbDevAttributesScope: "$" + DbDevAttributesScope,
+				},
+				DbCount: bson.M{
+					"$sum": 1,
+				},
+			},
```

Comment:
```suggestion
			"$group": bson.D{
				{Key: "_id", Value: nil},
				{Key: DbDevAttributesName, Value:  "$" + DbDevAttributesName},
				{Key: DbDevAttributesScope, Value: "$" + DbDevAttributesScope},
				DbCount: bson.M{
					"$sum": 1,
				},
			},
```
This way you don't have to do so many allocations when decoding the result.

Reply:
Well, this does not work: in mongodb, only the `_id` field is used for grouping fields, all other fields need to specify an accumulator (https://docs.mongodb.com/manual/reference/operator/aggregation/group/). Applying your change, I get:

The field 'name' must be an accumulator object

Reply:
I've done it using an extra step in the pipeline using $project, which allows us to deserialize the cursor directly to the array of model.FilterAttribute.

Reply:
Ahh, ok. I see I misunderstood the docs for `$group`.
I guess another option without adding an extra step to the pipeline would be creating an alias
```
type MongoFilterAttr model.FilterAttribute

func (f *MongoFilterAttr) UnmarshalBSON(b []byte) error {
       ... 
} 
```